src/session.c

```diff
@@ -775,15 +770,6 @@ static void construct_resolve_slist(GHashTable *addr_table, bool new_session) {
 
         // We need to sort on health score, but use the addr name.
         addr_score[addr_score_idx] = g_new(struct addr_score_s, 1);
-        // Take the opportunity to decrement the score by the amount of time which has passed since it last went bad.
-        // This will update the hashtable entry. It's a pointer, so update will stick
-        if (!new_session && healthstatus->score != HEALTHY) {
-            --healthstatus->score;
```

Comment:
I removed it intentionally, and had an iron-clad line of reasoning. Alas, on reflection, there's a chink in the clad iron. If we get a new session, it means we update the health status, and we don't want to undo that update here, so we only do it when we don't get a new session. That used to mean timeout. The chink in the iron is that we now don't update status on timeout. Let me think again on this one. Is 10 minutes too long an interval between health updates? Should we have an update timer (say 2 minutes as before). I'll ponder.